### PR TITLE
add request test, use const for http status

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* Add another test for rotation
* Replace http code number with consts for readability

### CODE REVIEW

File: internal/export/batcher.go [hunk 1]
```diff
@@ -94,11 +94,11 @@ func (s *Server) handleCreateBatches() http.Handler {
 			switch {
 			case errors.Is(err, context.DeadlineExceeded):
 				logger.Infow("batch creation timed out")
-				w.WriteHeader(200)
+				w.WriteHeader(http.StatusOK)
```
Comment (whaught) [inline internal/export/batcher.go:97]: @mikehelmick these don't seem like the right statuses for these?

File: internal/export/batcher.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func (s *Server) handleCreateBatches() http.Handler {
 			if errors.Is(err, database.ErrAlreadyLocked) {
 				stats.Record(ctx, mBatcherLockContention.M(1))
 				logger.Infow("already locked")
-				w.WriteHeader(200)
+				w.WriteHeader(http.StatusOK)
```
Comment (sethvargo) [inline internal/export/batcher.go:54]: I actually have a note to go through and make helpers for these and always return JSON.

200 is the right code though I think. It means someone else is already running. We don't want Cloud Scheduler to retry in this case.

File: internal/keyrotation/rotate_test.go [hunk 3]
```diff
@@ -32,6 +34,42 @@ import (
...
+	}
+
+	env := serverenv.New(ctx, serverenv.WithKeyManager(kms), serverenv.WithDatabase(testDB))
+	server, err := NewServer(config, env)
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	handler := server.handleRotateKeys()
+
+	r := httptest.NewRequest("GET", "/", nil)
```
Comment (sethvargo) [inline internal/keyrotation/rotate_test.go:59]: Use NewRequestWithContext instead of New and then Clone

**Review by sethvargo — CHANGES_REQUESTED** [review]

File: internal/export/batcher.go (hunk 1 above)
Comment (mikehelmick) [reply internal/export/batcher.go:97]: this one should probably be 500

File: internal/export/batcher.go (hunk 2 above)
Comment (whaught) [reply internal/export/batcher.go:54]: Agreed on that one (although maybe we should comment that it's for scheduler - in normal REST world I might expect 409 StatusConflict or something).

Comment (whaught) [reply internal/export/batcher.go:54]: but what about the above in the export batcher enum?

Comment (mikehelmick) [reply internal/export/batcher.go:54]: right, this is 200 so the CS doesn't report failure, I know it's a bit weird

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**